hashintel/hash
# H-934, H-1025, H-1283, H-1445: Use new inference endpoint in browser plugin, add passive inference

## 🌟 What is the purpose of this PR?

Update the `plugin-browser` to:
- Use the new inference endpoint added in #3630 
- Allow/track multiple inference requests
- Add passive inference option

**Removes**: the AI Python worker and the GraphQL `inferEntities` mutation from the previous implementation

## 🔍 What does this change?



As well as the above:
- extracts a shared `pluralize` function as I needed to correct a pluralization (which we might not need if we decide not to pluralize link titles such as Employed By, but it'll be useful when we need to correct others)
- updates some types to do with inference to add more useful bits of information

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:



### 📜 Does this require a change to the docs?




The changes in this PR:



### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## 🐾 Next steps

- H-1195: Have the inference feature look for existing entities to update (rather than always creating new)
- H-1284: Allow limiting passive inference to specific domains
- H-1446: Test with more complex types
- Next priorities to be determined by testing – might be better error handling (e.g. ask the model to retry validation failures), or something else



## ❓ How to test this?



1. See #3630 for instructions on running the AI TS worker locally
2. Run `yarn dev` in the plugin, then in chrome://extensions 'load unpacked' and select the `build` directory

## 📹 Demo





https://github.com/hashintel/hash/assets/37743469/0d906539-0329-46ee-a739-89cfc8af5b58
